### ISSUE TITLE
Sorting menus by the order they're placed

### ISSUE THREAD

**Issue by chris-79** [gohugoio/hugo#10710]
Why do (menu) items have to be sorted alphabetically by default? And why do we **_have to_** use a "weight" parameter when we can just put them in the order we want??

I just don't understand the reasoning behind this _at all_. &nbsp; ELI5

This should show up in the order that it appears, but it doesn't… it gets sorted alphabetically ("Directory" is first)

```
menu:
  siteNavigation:
    - name: Testing 1-2-3
      url: /123
    - name: Directory
      url: /directory
```

```
{{ with .Site.Menus.sitenavigation }}
<nav aria-label="Primary">
  <ul>
    {{ range . -}}
    <li><a href="{{ .URL }}">
      {{- (replace .Name " " "&nbsp;") | safeHTML -}}
    </a></li>
    {{ end -}}
  </ul>
</nav>
{{ end -}}
```

**Comment by jmooring** [comment]
I understand what you want, but it gets a bit complicated. You can define menu entries:

- In site configuration
- In front matter
- By using sectionPagesMenu = "main" in site configuration

In this example, we use all three methods...

content structure

```text
content/
├── articles/
│   ├── article-1.md
│   └── article-2.md
├── books/
│   ├── book-1.md
│   └── book-2.md
├── posts/
│   ├── post-1.md
│   └── post-2.md
└── _index.md
```

site config

```
sectionPagesMenu = "main"

[[menu.main]]
name = 'Home'
pageRef = '/'

[[menu.main]]
name = 'Article 2'
pageRef = '/articles/article-2'
```

content/books/book-2.md

```text
+++
title = 'Book 2'
date = 2023-02-09T08:10:40-08:00
draft = false
menu = 'main'
+++
```

template

```
{{ range site.Menus.main }}
  <a href="{{ .URL }}">{{ .Name }}</a><br>
{{ end }}
```

rendered

```
Article 2
Articles
Book 2
Books
Home
Posts
```

How should those be sorted in the absence of weights?

**Comment by chris-79** [comment]
imo, in the hierarchical order in which they exist… Site config is "highest" and first, then, as they appear from the top-level on down, they get appended to the end 🤷‍♂️

I definitely see why weights are useful in such situations, but for basic listings without this kind of mixing, there should be some option to just accept them in the order they are set

I'm no genius when it comes to actually coding these things, obviously, but it's frustrating when we take the time to put things in the order we want them, just for them to be ignored in the end

**Comment by bep** [comment]
Since we're merging different menu sources (site config(s), theme configs, page front matter) and we rebuild the menu whenever one of these changes, we need a defined sort order. What you ask for would maybe be possible, but I'm not sure how useful it would be beyond the simple setup -- and in that simple setup it should also be trivial to add some weight attributes.

**Comment by yaardie** [comment]
Adding menu weights is not a tall order. My suggestion is to close this issue and retain the default behaviour.
